javacript-exercise/formValidation.js

```diff
@@ -0,0 +1,119 @@
+function comparePasswords() {
+  const passwordElement = document.getElementById('password');
+  const passwordConfirmElement = document.getElementById('password-confirmation');
+  const errorMessage2 = document.querySelector(
+    '.form-message[data-attribute="password-confirmation"]'
+  );
+
+  if(passwordElement.value !== passwordConfirmElement.value) {
+    errorMessage2.textContent = "mat khau khong trung nhau!";
+
+    return false;
+
+  } else {
+    errorMessage2.textContent = '';
+  }
+
+  return true;
+}
+
+function displayErrorMessage(field, msg) {
+  const errorMessage = document.querySelector(
+    `.form-message[data-attribute="${field}"]`
+  );
+  errorMessage.textContent = msg;
+}
+
+function checkEmpty(fieldName) {
+  const inputElement = document.getElementById(fieldName);
+  if (inputElement.value === "") {
+    displayErrorMessage(fieldName, "nhap thieu roi hyhy");
+
+    return false;
+  } else {
+    displayErrorMessage(fieldName, "");
+  }
+
+  return true;
+}
+
+function checkEmail() {
+  const emailInputElement = document.getElementById("email");
+  const emailPattern = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})$/;
+
+  if (!emailPattern.test(emailInputElement.value)) {
+    displayErrorMessage('email', 'Email khong hop le');
+    
+    return false;
+  } else {
+    displayErrorMessage('email', '');
+  }
+
+  return true;
+}
+
+function checkBox() {
+  const check1Element = document.getElementById('check1');
+  const check2Element = document.getElementById('check2');
+
+  if (!check1Element.checked && !check2Element.checked) {
+    displayErrorMessage('checkbox', 'Vui long check');
+
+    return false;
+  } else {
+    displayErrorMessage('checkbox', '');
+  }
+
+  return true;
+}
+
+function checkSelector() {
+  const checkSelectorElement = document.getElementById('country');
+
+  if (checkSelectorElement.value === ''){
+    displayErrorMessage('country', 'Vui long chon quoc gia');
+
+    return false;
+  } else {
+    displayErrorMessage('country', '');
+  }
+
+  return true;
+}
+
+const formElement = document.querySelector(".form");
+formElement.addEventListener("submit", (e) => {
+  const checkings = [];
+  
+  checkings.push(checkEmpty('fullname'));
+  checkings.push(checkEmail());
+  checkings.push(checkEmpty('password'));
+  checkings.push(checkEmpty('password-confirmation'));
```

Comment:
Should group all the validations of for value into a function. In this function, validate form field values and patch the errors into formError
formError should be object like:

{
  fullname: 'Name is required',
  email: 'Invalid email',
....
}

if field's value is valid, remove this field from formError
if field's value is invalid, add this field error to formError

then use formError to check if the form is valid or not